packages/core/content-releases/admin/src/pages/ReleaseDetailsPage.tsx

```diff
@@ -21,69 +34,343 @@ import styled from 'styled-components';
-  const days = 0; // TODO: replace it with the number of days since the release was created
-  const createdBy = 'John Doe'; // TODO: replace it with the name of the user who created the release
+  const title = data?.data?.name;
+  const createdAt = data?.data?.createdAt;
+  const totalEntries = data?.data?.actions?.meta?.count || 0;
+  const { unit, value } = createdAt ? countDays(createdAt) : { unit: 'day', value: 0 };
+
+  const createdBy = `${data?.data?.createdBy?.firstname} ${data?.data?.createdBy?.lastname}` || '';
 
   const handleTogglePopover = () => {
     setIsPopoverVisible((prev) => !prev);
   };
 
-  const toggleEditReleaseModal = () => {
-    setReleaseModalShown((prev) => !prev);
-  };
-
   const openReleaseModal = () => {
     toggleEditReleaseModal();
     handleTogglePopover();
   };
 
-  const [updateRelease, { isLoading }] = useUpdateReleaseMutation();
+  return (
+    <Main aria-busy={isLoadingDetails}>
+      <Box paddingBottom={8}>
+        <HeaderLayout
+          title={title}
+          subtitle={
+            !isLoadingDetails &&
+            formatMessage(
+              {
+                id: 'content-releases.pages.Details.header-subtitle',
+                defaultMessage: '{number, plural, =0 {No entries} one {# entry} other {# entries}}',
+              },
+              { number: totalEntries }
+            )
+          }
+          navigationAction={
+            <Link startIcon={<ArrowLeft />} to="/plugins/content-releases">
+              {formatMessage({
+                id: 'global.back',
+                defaultMessage: 'Back',
+              })}
+            </Link>
+          }
+          primaryAction={
+            <Flex gap={2}>
+              <IconButton
+                label={formatMessage({
+                  id: 'content-releases.header.actions.open-release-actions',
+                  defaultMessage: 'Release actions',
+                })}
+                ref={moreButtonRef}
+                onClick={handleTogglePopover}
+              >
+                <More />
+              </IconButton>
+              {isPopoverVisible && (
+                <Popover
+                  source={moreButtonRef}
+                  placement="bottom-end"
+                  onDismiss={handleTogglePopover}
+                  spacing={4}
+                  minWidth="242px"
+                >
+                  <Flex alignItems="center" justifyContent="center" direction="column" padding={1}>
+                    <CheckPermissions permissions={PERMISSIONS.update}>
+                      <PopoverButton
+                        paddingTop={2}
+                        paddingBottom={2}
+                        paddingLeft={4}
+                        paddingRight={4}
+                        alignItems="center"
+                        gap={2}
+                        as="button"
+                        hasRadius
+                        onClick={openReleaseModal}
+                      >
```

Comment:
I guess this could be extracted into a component to avoid duplicating the same thing below